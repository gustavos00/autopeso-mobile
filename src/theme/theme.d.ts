import "styled-components"
import { ThemeInterface } from "./index";

type ThemeInterface = typeof theme

declare module "styled-components" {
    interface DefaultTheme extends ThemeInterface {}
}