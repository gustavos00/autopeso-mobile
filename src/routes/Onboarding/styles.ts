import styled from 'styled-components/native';

interface x {
    isActive: boolean
}

export const Container = styled.ScrollView`
    flex: 1;
    padding-right: 10%;
    padding-left: 10%;
    background-color: ${props => props.theme.colors.black.main};
`;

export const Title = styled.Text`
    margin-bottom: ${props =>props.theme.metrics.sm}px;

    background-color: ${props => props.theme.colors.black.main};

    color: ${props => props.theme.colors.white.main};
    font-size: 32px;
    font-weight: 600;
    text-align: center;
`;

export const Description = styled.Text`
    margin-bottom: ${props => props.theme.metrics.xlg}px;

    background-color: ${props => props.theme.colors.black.main};

    color: ${props => props.theme.colors.gray.main};
    font-size: 14px;
    text-align: center;
`;

export const DotsContainer = styled.View`
    margin-bottom: ${props => props.theme.metrics.xlg}px;
    flex-direction: row;
`;

export const Dot = styled.View<x>`
    width: ${props => props.isActive ? props.theme.metrics.xlg : props.theme.metrics.md}px;
    height: ${props => props.theme.metrics.md}px;
    margin-left: 5px;
    margin-right: 5px;

    border-radius: 100px;
    background-color: ${props => props.isActive ? props.theme.colors.gray.main : props.theme.colors.gray.secundary};
`;


