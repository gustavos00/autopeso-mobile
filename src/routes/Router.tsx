import React from 'react'

import { NavigationContainer } from '@react-navigation/native'
import { createNativeStackNavigator } from '@react-navigation/native-stack'

import { Onboarding } from './Onboarding'

type RootStackParamList = {
  Onboarding: undefined
}

const { Navigator, Screen } = createNativeStackNavigator<RootStackParamList>()

export function Router(): JSX.Element {
  return (
      <NavigationContainer>
        <Navigator
          initialRouteName="Onboarding"
          screenOptions={{
            headerShown: false
          }}>
            <Screen name="Onboarding" component={Onboarding} />
        </Navigator>
      </NavigationContainer>
  )
}
