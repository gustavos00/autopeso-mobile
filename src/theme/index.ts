interface ColorsContent {
    main: string;
    secundary?: string;
    gradient?: string
}

interface ColorsStructure {
    primary: ColorsContent;
    black: ColorsContent;
    white: ColorsContent;
    gray: ColorsContent;
}

interface MetricsStructure {
    sm: number,
    md: number,
    lg: number,
    xlg: number,
}

export interface ThemeInterface {
    colors: ColorsStructure
    metrics: MetricsStructure
}

export const theme: ThemeInterface = {
    colors: {    
        primary: {
            main: '#F24462'
        },
        black: {
            main: '#252732',
            secundary: '#2D324B'
        },
        white: {
            main: '#fff',
        },
        gray: {
            main: '#D9D9D9',
            secundary: '#A3A3A3'
        }
    },
    metrics: {
        sm: 8,
        md: 16,
        lg: 24,
        xlg: 32,
    }
} 