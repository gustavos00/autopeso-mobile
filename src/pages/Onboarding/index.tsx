import React from 'react'
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { useNavigation } from '@react-navigation/native';

import onboarding from '../../static/onboarding.json'
import { Button } from '../../components/Button'

import * as S from './styles'

export function Onboarding(): JSX.Element {
  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  
  return (
    <S.Container 
      contentContainerStyle={{
        flex:1,
        justifyContent: 'center', 
        alignItems: 'center'
      }}>
      <S.BoxContainer 
        pagingEnabled
        horizontal
        scrollEventThrottle={32}
        showsHorizontalScrollIndicator={false}  
        contentContainerStyle={{alignItems: 'center'}}
      >
        <S.Box>

        </S.Box>

        <S.Box>

        </S.Box>

        <S.Box>

        </S.Box>
      </S.BoxContainer>

      <S.Title>{onboarding.title}</S.Title>
      <S.Description>{onboarding.description}</S.Description>

      <S.DotsContainer>
        <S.Dot isActive={false}/>
        <S.Dot isActive={false} />
        <S.Dot isActive={true} />
      </S.DotsContainer>

      <Button 
        text={'Get Started'}
        handleOnTouch={() => navigation.navigate('SignIn')}
      />
    </S.Container>
  )
}
