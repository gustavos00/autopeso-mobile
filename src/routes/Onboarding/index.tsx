import React from 'react'
import { useNavigation } from '@react-navigation/native';

import onboarding from '../../static/onboarding.json'
import { Button } from '../../components/Button'

import * as S from './styles'


export function Onboarding(): JSX.Element {
  const navigation = useNavigation();
  
  return (
    <S.Container 
      contentContainerStyle={{
        flex:1,
        justifyContent: 'center', 
        alignItems: 'center'
      }}>

      <S.Title>{onboarding.title}</S.Title>
      <S.Description>{onboarding.description}</S.Description>

      <S.DotsContainer>
        <S.Dot isActive={false}/>
        <S.Dot isActive={false} />
        <S.Dot isActive={true} />
      </S.DotsContainer>

      <Button 
        text={'Get Started'}
        handleOnTouch={() => navigation.navigate('signIn')}
      />
    </S.Container>
  )
}
