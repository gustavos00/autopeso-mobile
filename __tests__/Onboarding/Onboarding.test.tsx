import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import { Onboarding } from '../../src/routes/Onboarding';
import { theme } from '../../src/theme/index'

jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: jest.fn(() => ({}))
  };
});

describe('Onboarding Screen', () => {
  it('should render Layout with all the content', async () => {
      const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <Onboarding />
      </ThemeProvider>).toJSON();
      expect(tree).toMatchSnapshot();
    });
});