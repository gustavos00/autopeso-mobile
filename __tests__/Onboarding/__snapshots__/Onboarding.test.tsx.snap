// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding Screen should render Layout with all the content 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  style={
    [
      {
        "backgroundColor": "#252732",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingLeft": "10%",
        "paddingRight": "10%",
      },
    ]
  }
>
  <View>
    <Text
      style={
        [
          {
            "backgroundColor": "#252732",
            "color": "#fff",
            "fontSize": 32,
            "fontWeight": "600",
            "marginBottom": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Car Wash
    </Text>
    <Text
      style={
        [
          {
            "backgroundColor": "#252732",
            "color": "#D9D9D9",
            "fontSize": 14,
            "marginBottom": 32,
            "textAlign": "center",
          },
        ]
      }
    >
      What you think about giving your car the service that he deserve?
    </Text>
    <View
      style={
        [
          {
            "flexDirection": "row",
            "marginBottom": 32,
          },
        ]
      }
    >
      <View
        isActive={false}
        style={
          [
            {
              "backgroundColor": "#A3A3A3",
              "borderRadius": 100,
              "height": 16,
              "marginLeft": 5,
              "marginRight": 5,
              "width": 16,
            },
          ]
        }
      />
      <View
        isActive={false}
        style={
          [
            {
              "backgroundColor": "#A3A3A3",
              "borderRadius": 100,
              "height": 16,
              "marginLeft": 5,
              "marginRight": 5,
              "width": 16,
            },
          ]
        }
      />
      <View
        isActive={true}
        style={
          [
            {
              "backgroundColor": "#D9D9D9",
              "borderRadius": 100,
              "height": 16,
              "marginLeft": 5,
              "marginRight": 5,
              "width": 32,
            },
          ]
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#F24462",
          "borderRadius": 8,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#fff",
              "fontSize": 16,
              "fontWeight": "600",
            },
          ]
        }
      >
        Get Started
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
